### PR TITLE
DON'T Merge! Don't

### PR DESCRIPTION
**Pull request by akaala**
Feature Test: meta server.

### CODE REVIEW

File: hermes-core/src/main/java/com/ctrip/hermes/core/bo/HostPort.java [hunk 1]
```diff
@@ -30,4 +30,30 @@ public void setPort(int port) {
...
+		HostPort hostPort = (HostPort) o;
+
+		if (m_port != hostPort.m_port) return false;
+		return !(m_host != null ? !m_host.equals(hostPort.m_host) : hostPort.m_host != null);
+
+	}
+
+	@Override
+	public int hashCode() {
+		int result = m_host != null ? m_host.hashCode() : 0;
```
Comment (yiming187) [inline hermes-core/src/main/java/com/ctrip/hermes/core/bo/HostPort.java]: Use Objects.hashCode and Objects.equals

File: hermes-core/src/main/java/com/ctrip/hermes/core/lease/Lease.java [hunk 2]
```diff
@@ -2,9 +2,11 @@
 
 import java.util.concurrent.atomic.AtomicLong;
 
+
 import com.alibaba.fastjson.annotation.JSONField;
 import com.ctrip.hermes.core.service.SystemClockService;
 import com.ctrip.hermes.core.utils.PlexusComponentLocator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
```
Comment (yiming187) [inline hermes-core/src/main/java/com/ctrip/hermes/core/lease/Lease.java]: why do we need two json serialization systems

File: hermes-core/pom.xml [hunk 3]
```diff
@@ -27,6 +27,13 @@
 			<groupId>org.slf4j</groupId>
 			<artifactId>slf4j-log4j12</artifactId>
 		</dependency>
+
+        <!--for Lease.class: JsonIgnore-->
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-annotations</artifactId>
+        </dependency>
```
Comment (leoliang) [inline hermes-core/pom.xml]: why do we need this dependency?

File: hermes-core/src/main/java/com/ctrip/hermes/core/lease/Lease.java [hunk 4]
```diff
@@ -39,11 +41,13 @@ public long getExpireTime() {
 		return m_expireTime.get();
 	}
 
+	@JsonIgnore
```
Comment (leoliang) [inline hermes-core/src/main/java/com/ctrip/hermes/core/lease/Lease.java]: we use fast json only

File: hermes-metaserver/src/main/java/com/ctrip/hermes/metaserver/cluster/ClusterStateHolder.java [hunk 5]
```diff
@@ -62,7 +62,7 @@ public void notLeader() {
 
 			@Override
 			public void isLeader() {
-				log.info("Become leader");
+				log.info("{} Become leader", m_config.getMetaServerPort());
```
Comment (leoliang) [inline hermes-metaserver/src/main/java/com/ctrip/hermes/metaserver/cluster/ClusterStateHolder.java]: remove this code if it's for test case only

File: hermes-metaserver/src/main/java/com/ctrip/hermes/metaserver/commons/BaseLeaseHolder.java [hunk 6]
```diff
@@ -142,7 +145,12 @@ private LeaseContext getLeaseContext(Key contextKey) {
 		m_LeaseContextsLock.writeLock().lock();
 		try {
 			if (!m_LeaseContexts.containsKey(contextKey)) {
+
+				//todo: 解决为什么这里的pair会被put2次，输出pari看看
+
 				m_LeaseContexts.put(contextKey, new LeaseContext());
+
```
Comment (leoliang) [inline hermes-metaserver/src/main/java/com/ctrip/hermes/metaserver/commons/BaseLeaseHolder.java]: remove this code if it's for test case only

File: hermes-metaserver/src/main/java/com/ctrip/hermes/metaserver/commons/BaseLeaseHolder.java [hunk 7]
```diff
@@ -133,6 +133,9 @@ private void removeExpiredLeases(Key contextKey, Map<String, ClientLeaseInfo> ex
 			Entry<String, ClientLeaseInfo> entry = iter.next();
 			Lease lease = entry.getValue().getLease();
 			if (lease != null && lease.isExpired()) {
+
+				log.info("==Remove: {}, lease: {}, isExpired: {}, getRemainingTime: {}",
```
Comment (leoliang) [inline hermes-metaserver/src/main/java/com/ctrip/hermes/metaserver/commons/BaseLeaseHolder.java]: remove this code if it's for test case only

File: hermes-metaserver/src/main/java/com/ctrip/hermes/metaserver/event/impl/FollowerInitEventHandler.java [hunk 8]
```diff
@@ -119,7 +119,7 @@ private Meta fetchMetaInfo(MetaInfo metaInfo) {
 			}
 
 		} catch (Exception e) {
-			log.error("Failed to fetch meta from leader({}:{})", metaInfo.getHost(), metaInfo.getPort(), e);
+			log.error("Failed to fetch meta from leader({}:{})", metaInfo.getHost(), metaInfo.getPort());
```
Comment (leoliang) [inline hermes-metaserver/src/main/java/com/ctrip/hermes/metaserver/event/impl/FollowerInitEventHandler.java]: why remove exception info in log?

File: hermes-metaserver/src/main/java/com/ctrip/hermes/metaserver/meta/DefaultMetaServerAssigningStrategy.java [hunk 9]
```diff
@@ -28,7 +28,10 @@
 			int metaServerPos = 0;
 			int metaServerCount = metaServers.size();
 
-			if (topics != null) {
+			/**
+			 * add "&& metaServerCount > 0". If there is no metaServers running.
+			 */
+			if (topics != null && metaServerCount > 0) {
```
Comment (leoliang) [inline hermes-metaserver/src/main/java/com/ctrip/hermes/metaserver/meta/DefaultMetaServerAssigningStrategy.java]: no metaServer is running? At least the instance which is running this code is a running meta server instance

**Pull request closed without merging**